### ISSUE TITLE
Zoom limit of 700% should be user-configurable

### ISSUE THREAD

**Issue by yurivict** [xournalpp/xournalpp#3555]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: FreeBSD
 - (Linux only) Desktop environment: xfce4
 - Version of Xournal++: 1.1.0

**Describe the bug**
Max zoom percentage is 700%. I happen to have a one-page pdf with a lot of elements in it. In order to see small details zoom > 700% is needed but this can't be currently done.

I suggest to make this limit configurable in the Zoom page in the Preferences window.

**Comment by bhennion** [comment]
With the current rendering system, removing the cap on Zoom ratios would open Pandora's box: there are a lot of high zoom related lags that need a (slowly ongoing) rewrite of the rendering system.

Until this is done, I don't think removing the limit is a good idea.

**Comment by m-GDEV** [comment]
Any update this issue?

**Comment by rolandlo** [comment]
Not really. I think we can close the issue for now.

**Comment by twitwi** [comment]
In case it can be a workaround for people, I encountered the same limitation and ended up pre-scaling the pdf, inspired by https://www.baeldung.com/linux/pdf-change-page-size#cli-tools-to-convert-pdf-size
I used, to pre-scale by 5 (500%)

```bash
s=$(pdfinfo MY.pdf | grep 'Page size' | awk '{printf "{%.2fpt,%.2fpt}\n", $3*5, $5*5}')
pdfjam --papersize "$s" MY.pdf --outfile OUT.pdf
```

**Comment by vigiraud** [comment]
I'm using a graphic tablet to correct papers, and a zoom level above 700% would be very useful to make annotations in margins.

**Comment by disposableone** [comment]
Yes, need this too. 

We don't even care about any lag! We are not BROWSING stuff on that zoom. We zoom in to write something, then zoom back out, for example.

**Comment by aisandbox2026** [comment]
Xournal++ 4000% Zoom Mod — Windows 11

Tested on:
Xournal++ 1.3.6
Windows 11
Built: Jul 13, 2026
GTK: 3.24.52
Git commit: 0537e885

I wanted more than the built-in 700% zoom limit, so I hex-edited the installed Xournal++ executable.

For this exact build, the max zoom value is stored at:

Offset:
0x2F40F0

Original value:
00 00 00 00 00 00 1C 40

This is the little-endian IEEE-754 double for:
7.0 = 700%

I changed it to:

00 00 00 00 00 00 44 40

This is:
40.0 = 4000%

What I did:

1. Made a backup of the original xournalpp.exe.
2. Opened xournalpp.exe in a hex editor.
3. Went to offset 0x2F40F0.
4. Changed:

   00 00 00 00 00 00 1C 40

   to:

   00 00 00 00 00 00 44 40

5. Saved the modified copy as:

   xournalpp - Copy.exe

6. Left it in the same folder as the original Xournal++ executable.
7. Made a shortcut to the modified EXE and named it XOURNAL4000.

Result:

Xournal++ now zooms up to 4000%.

The zoom display/slider still visually stops at 700%, but once you reach 700%, just keep scrolling and it continues zooming beyond that.

On my HP EliteBook x360 1030 G3 with an 8th-gen Intel Core i7 and 16 GB RAM, there is some slowdown at very high zoom levels, but it is still completely usable.

Important:

The offset 0x2F40F0 is confirmed for the Xournal++ 1.3.6 Windows build listed above. Other versions/builds may store the value at a different offset.

The value to look for is:

00 00 00 00 00 00 1C 40

which represents 7.0 / 700%.

Other useful values:

10.0 = 1000%
00 00 00 00 00 00 24 40

20.0 = 2000%
00 00 00 00 00 00 34 40

40.0 = 4000%
00 00 00 00 00 00 44 40

Always keep a backup of the original EXE before editing.

Btw it took like 5min if that. 

HOPE THIS HELPS SOMEONE!
